mistralrs-paged-attn/src/metal/kernels/mod.rs

```diff
@@ -51,13 +51,19 @@ pub struct Kernels {
     pipelines: RwLock<Pipelines>,
 }
 
-impl Default for Kernels {
-    fn default() -> Self {
-        Self::new()
-    }
-}
+static INIT: Once = Once::new();
+pub(crate) static mut G_KERNEL: Option<Kernels> = None;
 
 impl Kernels {
+    pub fn default() -> &'static Option<Self> {
+        unsafe {
+            INIT.call_once(|| {
+                G_KERNEL = Some(Self::new());
+            });
+            &G_KERNEL
```

Comment:
On Rust 1.83, I'm getting: `warning: creating a shared reference to mutable static is discouraged`.

Perhaps we can make it a Mutex?

Reply:
Warning fixed with once_cell.